src/libcollections/slice.rs

```diff
@@ -200,28 +200,71 @@ impl<T> [T] {
         core_slice::SliceExt::first(self)
     }
 
-    /// Returns a mutable pointer to the first element of a slice, or `None` if it is empty
+    /// Returns a mutable pointer to the first element of a slice, or `None` if it is empty.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let x = &mut [0, 1, 2];
+    ///
+    /// if let Some(first) = x.first_mut() {
+    ///     *first = 5;
+    /// }
+    /// assert_eq!(x, &[5, 1, 2]);
+    /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn first_mut(&mut self) -> Option<&mut T> {
         core_slice::SliceExt::first_mut(self)
     }
 
     /// Returns the first and all the rest of the elements of a slice.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let x = &[0, 1, 2];
+    /// if let Some((first, elements)) = x.split_first() {
```

Comment:
can you add a newline above the `if`, like you did in the first example? Same with tons of them below.